contracts/zero-ex/tests/Multiplex.t.sol

```diff
@@ -0,0 +1,1453 @@
+                        sellAmount: sellAmount,
+                        data: hex""
+                    })
+                ),
+                sellAmount,
+                0
+            )
+        {
+            fail("did not revert");
+        } catch Error(string memory reason) {
+            assertEq(reason, "MultiplexFeature::_executeBatchSell/INVALID_SUBCALL", "wrong revert reason");
+        } catch {
+            fail("low-level revert");
+        }
+    }
+
+    function test_multiplexBatchSellTokenForToken_3() public explain("reverts if the full sell amount is not sold") {
+        LibNativeOrder.RfqOrder memory rfqOrder = makeTestRfqOrder();
+        mintTo(address(rfqOrder.takerToken), rfqOrder.taker, rfqOrder.takerAmount);
+
+        try
+            zeroExDeployed.zeroEx.multiplexBatchSellTokenForToken(
+                rfqOrder.takerToken,
+                rfqOrder.makerToken,
+                makeArray(makeRfqSubcall(rfqOrder)),
+                rfqOrder.takerAmount + 1,
+                rfqOrder.makerAmount
+            )
+        {
+            fail("did not revert");
+        } catch Error(string memory reason) {
+            assertEq(reason, "MultiplexFeature::_executeBatchSell/INCORRECT_AMOUNT_SOLD", "wrong revert reason");
+        } catch {
+            fail("low-level revert");
+        }
+    }
+
+    function test_multiplexBatchSellTokenForToken_4() public explain("RFQ, fallback(UniswapV2)") {
+        LibNativeOrder.RfqOrder memory rfqOrder = makeTestRfqOrder();
+        createUniswapV2Pool(uniV2Factory, dai, zrx, 10e18, 10e18);
+        mintTo(address(rfqOrder.takerToken), rfqOrder.taker, rfqOrder.takerAmount);
+
+        try
+            zeroExDeployed.zeroEx.multiplexBatchSellTokenForToken(
+                rfqOrder.takerToken,
+                zrx,
+                makeArray(
+                    makeRfqSubcall(rfqOrder),
+                    makeUniswapV2BatchSubcall(makeArray(address(dai), address(zrx)), rfqOrder.takerAmount, false)
+                ),
+                rfqOrder.takerAmount,
+                0
+            )
+        {
+            assertLogs(
+                makeArray(
+                    describeLog({
+                        topics: makeArray(RFQ_ORDER_FILLED_SIG),
+                        data: abi.encode(
+                            zeroExDeployed.features.nativeOrdersFeature.getRfqOrderHash(rfqOrder),
+                            rfqOrder.maker,
+                            rfqOrder.taker,
+                            rfqOrder.makerToken,
+                            rfqOrder.takerToken,
+                            rfqOrder.takerAmount,
+                            rfqOrder.makerAmount,
+                            rfqOrder.pool
+                        )
+                    })
+                )
+            );
+        } catch Error(string memory reason) {
+            fail("reverted");
+            fail(reason);
+        } catch {
+            fail("low-level revert");
+        }
```

Comment:
Why?

Just let the error propagate and the test will fail with a relevant message to the console

Reply:
This whole try/catch thing I'm doing is a sort of vestigial pattern left over from when I had the tests structured in a different way, and I neglected to realize it was no longer necessary after some rewriting. Definitely a good catch 😅